### ISSUE TITLE
Announcements

### ISSUE THREAD

**Issue by sameersbn** [sameersbn/docker-gitlab#39]
Subscribe to this issue to get notified about project announcements.
This is mainly created so that you can keep up with the changes in the application stack.
**Please do not use this thread to report issues.**

**Comment by sameersbn** [comment]
2014-02-07:
- The `GITLAB_SIGNUP`, `GITLAB_SIGNIN`, `GITLAB_PROJECTS_LIMIT` and `GITLAB_GRAVATAR_ENABLED` configuration parameters have been removed from the image. Since GitLab version `7.7`, these parameters can be set from the Administration page of the GitLab installation.

**Comment by sameersbn** [comment]
Folks using this `osx` + `boot2docker` + `sameersbn/postgresql`, it would be great to hear your inputs with issue https://github.com/sameersbn/docker-postgresql/issues/22. Thanks.

**Comment by sameersbn** [comment]
**2015-03-23**:
 - gitlab-ce version `7.9.0` is out now with BitBucket OAuth Support.
 - `GITLAB_RESTRICTED_VISIBILITY` has been removed and can now be configured from the UI
 - `SMTP_OPENSSL_VERIFY_MODE` now defaults to `peer`
 - `UNICORN_WORKERS` now defaults to `3`

**Comment by sameersbn** [comment]
**2015-04-22**:
- gitlab-ce `7.10.0` is out. This release removed options to configure external issue trackers through ENV variables. These settings have been moved to the `Settings` page of GitLab under the `Service Templates` section. If you are using an external issue tracker, don't forget to set it up.
- A new `GITLAB_EMAIL_REPLY_TO` option is added. You can configure the reply_to address in email sent out by gitlab.
- A new `LDAP_BLOCK_AUTO_CREATED_USERS` option is added for LDAP users

**Comment by sameersbn** [comment]
2015-05-11:
 - gitlab-ce `7.10.3` is out.
 - version `7.10.2` and above now has options to configure remote AWS backups.

**Comment by sameersbn** [comment]
**2014-09-24**
 - Since version `8.0.0`, Gitlab CE and CI are one. If you use GitLab CI, then please refer to https://github.com/sameersbn/docker-gitlab/blob/master/CI_MIGRATION.md for migration instructions

**Comment by sharpner** [comment]
you mean 2015 right? :)

**Comment by sameersbn** [comment]
fixed!

**Comment by sameersbn** [comment]
Since `8.0.4-1`, automated builds have been moved to https://quay.io/repository/sameersbn/gitlab. You will no longer find tagged versions of the image on the hub.docker.com. You can pull the image using `docker pull quay.io/sameersbn/gitlab:8.0.4-1`. hub.docker.com will only have builds tagged `latest`, no new versions will be tagged there.

The reason for this change is to that the release process on hub.docker.com is a very time consuming affair. I had to manually tag builds each time a release was made. quay.io allows me to push a git tag and it automatically builds the tag without any extra effort from me.

This does not change anything other than when you pull the image, you need to prefix `quay.io/` to the image image.

**Comment by pnelsonsr** [comment]
quay.io seems like an easier solution!

**Comment by sameersbn** [comment]
way better than hub.docker.com

 - simple release process
 - builds are faster
 - user can also download squashed docker image (experimental) or a rocket image
 - has a lot of statistics and visualization tools

**Comment by sharpner** [comment]
and a badge for the readme ;P

**Comment by sameersbn** [comment]
yea.. but it always says building, when its actually ready. i guess a github caching issue

**Comment by sameersbn** [comment]
Please be advised that http clones with relative_urls is broken in `8.1.0`.

**Comment by johnjelinek** [comment]
@sameersbn: what's an example where it would fail in `8.1.0`?

**Comment by sameersbn** [comment]
If `GITLAB_RELATIVE_URL_ROOT` is defined then cloning repos using http/https fails. SSH clones work fine.

**Comment by sameersbn** [comment]
release `8.1.0-1` fixes the https/http cloning issue when `GITLAB_RELATIVE_URL_ROOT` is used.

**Comment by davidwindell** [comment]
@sameersbn Docker Hub has just released dynamic matching of git branches and tags as well as faster builds.

**Comment by sameersbn** [comment]
@davidwindell has it been released? I don't see any options.

**Comment by sameersbn** [comment]
Builds are now back on Docker Hub and will also remain available on quay.io.

**Comment by sameersbn** [comment]
**2015-11-22**
- Gitlab 8.2.0 released
-  `&` characters in `GITLAB_TIMEZONE` are now automatically escaped. If you have manually escaped the `&` character with a `/` character then you need to remove it
- For the complete list of changes see https://github.com/sameersbn/docker-gitlab/blob/master/Changelog.md
- For the complete list of changes in the upstream GitLab project see https://gitlab.com/gitlab-org/gitlab-ce/blob/master/CHANGELOG

**Comment by taniadaniela** [comment]
@sameersbn I need the image of 7.8.3 version .. but I didn't find it in the tags.. :(

**Comment by obi12341** [comment]
@taniadaniela that is not related to any announcement. Maybe you open a new issue for that.

**Comment by mikehaertl** [comment]
@sameersbn No more announcements here? ;)

**Comment by sameersbn** [comment]
@mikehaertl this thread is intended to announce changes that could break existing deployments. recently there haven't been any such changes while new releases are being made :smile:

**Comment by mikehaertl** [comment]
@sameersbn Hmm, it's just that some of the previous announcements were about new features, too. So I though, you'd also announce new releases here. Unfortunately github doesn't have such a feature.

I think no one would be offended of an announcement too much ;)

**Comment by sameersbn** [comment]
@mikehaertl 

> I think no one would be offended of an announcement too much ;)

Agreed. Issue really is GitLab releases 3 - 4 releases a month. Updating this thread for every release becomes an issue at that point.

p.s. You can get notified about new releases using this atom feed https://github.com/sameersbn/docker-gitlab/tags.atom. If this does not work for you, then I will try updating this thread for every monthly release that happens on 22nd of each month.

If this does not work

**Comment by sameersbn** [comment]
Note that the release with GitLab 8.6.0 could be delayed by a few days as it breaks relative_url support in the image. If you are eager to use the latest release and **DO NOT** use relative_urls (i.e. `GITLAB_RELATIVE_URL_ROOT`) for your gitlab installation, then you can use the `latest` build of the image. The `8.6.0` tag will be delayed until relative_url support has been fixed.

Also note, GitLab `8.6.0` requires that `pg_trgm` extension is loaded for the gitlab database. If you are using the `sameersbn/postgresql` image then please upgrade to `sameersbn/postgresql:9.4-17` and add `DB_EXTENSION=pg_trgm` to the env of the postgresql container (eg, https://github.com/sameersbn/docker-gitlab/blob/master/docker-compose.yml#L8)

**Comment by lenovouser** [comment]
@sameersbn I have the `GITLAB_RELATIVE_URL_ROOT=` in my `docker-compose.yml` but it is empty, exactly like I posted it. Can I then upgrade without problems?

**Comment by sameersbn** [comment]
@lenovouser you are not using relative urls. you can upgrade.

**Comment by M0E-lnx** [comment]
I'm also waiting on the fix for relative root for 8.6.0.  Managed to get past the DB_EXTENSION by exec-ing into the postgresdb container, but still can't use it because of the relative root problem.

**Comment by sameersbn** [comment]
8.6.1 has been released with the relative_url fix.
builds status: https://hub.docker.com/r/sameersbn/gitlab/builds/

**Comment by djskinner** [comment]
> PostgreSQL users should also note that GitLab 8.6.0 requires that pg_trgm extension is loaded for the gitlab database. If you are using the sameersbn/postgresql image then please upgrade to sameersbn/postgresql:9.4-17 and add DB_EXTENSION=pg_trgm to the env of the postgresql container

This worked for me. I think it would be worthwhile adding this to the upgrade notes.

**Comment by sameersbn** [comment]
@djskinner done. thanks for the suggestion.

**Comment by sameersbn** [comment]
`sameersbn/gitlab:8.7.0` has been released. Please refer the [Changelog](../tree/master/Changelog.md).

IMPORTANT NOTICE: If you have reply by email enabled and use email sub-addressing please add `+%{key}` before the user part of the `GITLAB_INCOMING_EMAIL_ADDRESS`, eg. `GITLAB_INCOMING_EMAIL_ADDRESS=reply+%{key}@example.com`. This was automatically added in earlier releases, but since the feature now supports "dedicated email address" and "catch-all mailbox" this is no longer an option. see #670.

Refer to http://doc.gitlab.com/ce/incoming_email/README.html for more details.

**Comment by sameersbn** [comment]
**2016-06-01**
Releases 8.8.0 thru 8.8.2 have been published. We decided to put off the container registry support in these releases until we have a cleaner integration and nicer docs

**Comment by SolidNerd** [comment]
**2016-08-23**
We released `sameersbn/gitlab:8.11.0` this has some important changes. It's now required to set the following environment variables. `GITLAB_SECRETS_OTP_KEY_BASE` and `GITLAB_SECRETS_SECRET_KEY_BASE`. We recommend that these environment variables should have a string length of 64 characters. Please backup these keys. They will be needed for encryption like `GITLAB_SECRETS_DB_KEY_BASE`. We hope that you enjoy this.

**Comment by drubin** [comment]
This doesn't seem to be used anymore, also RSS seems like a better way. 

Maybe we should recommend we tell users to subscribe to the RSS feed for the releases (that is what I do via slack integration)

https://github.com/sameersbn/docker-gitlab/releases.atom
